src/axes.js

```diff
@@ -1,168 +1,31 @@
-export function autoAxisTicks({x, y, fx, fy}, {x: xAxis, y: yAxis, fx: fxAxis, fy: fyAxis}) {
-  if (fxAxis) autoAxisTicksK(fx, fxAxis, 80);
-  if (fyAxis) autoAxisTicksK(fy, fyAxis, 35);
-  if (xAxis) autoAxisTicksK(x, xAxis, 80);
-  if (yAxis) autoAxisTicksK(y, yAxis, 35);
-}
-
-function autoAxisTicksK(scale, axis, k) {
-  if (axis.ticks === undefined) {
-    const interval = scale.interval;
-    if (interval !== undefined) {
-      const [min, max] = extent(scale.scale.domain());
-      axis.ticks = interval.range(interval.floor(min), interval.offset(interval.floor(max)));
-    } else {
-      const [min, max] = extent(scale.scale.range());
-      axis.ticks = (max - min) / k;
-    }
-  }
-  // D3’s ordinal scales simply use toString by default, but if the ordinal
-  // scale domain (or ticks) are numbers or dates (say because we’re applying a
-  // time interval to the ordinal scale), we want Plot’s default formatter.
-  if (axis.tickFormat === undefined && isOrdinalScale(scale)) {
-    axis.tickFormat = formatDefault;
-  }
-}
 
 // Mutates axis.{label,labelAnchor,labelOffset} and scale.label!
-export function autoScaleLabels(channels, scales, {x, y, fx, fy}, dimensions, options) {
-  if (fx) {
-    autoAxisLabelsX(fx, scales.fx, channels.get("fx"));
-    if (fx.labelOffset === undefined) {
-      const {facetMarginTop, facetMarginBottom} = dimensions;
-      fx.labelOffset = fx.axis === "top" ? facetMarginTop : facetMarginBottom;
-    }
-  }
-  if (fy) {
-    autoAxisLabelsY(fy, fx, scales.fy, channels.get("fy"));
-    if (fy.labelOffset === undefined) {
-      const {facetMarginLeft, facetMarginRight} = dimensions;
-      fy.labelOffset = fy.axis === "left" ? facetMarginLeft : facetMarginRight;
-    }
-  }
-  if (x) {
-    autoAxisLabelsX(x, scales.x, channels.get("x"));
-    if (x.labelOffset === undefined) {
-      const {marginTop, marginBottom, facetMarginTop, facetMarginBottom} = dimensions;
-      x.labelOffset = x.axis === "top" ? marginTop - facetMarginTop : marginBottom - facetMarginBottom;
-    }
-  }
-  if (y) {
-    autoAxisLabelsY(y, x, scales.y, channels.get("y"));
-    if (y.labelOffset === undefined) {
-      const {marginRight, marginLeft, facetMarginLeft, facetMarginRight} = dimensions;
-      y.labelOffset = y.axis === "left" ? marginLeft - facetMarginLeft : marginRight - facetMarginRight;
-    }
-  }
-  for (const [key, type] of registry) {
-    if (type !== position && scales[key]) {
-      // not already handled above
-      autoScaleLabel(key, scales[key], channels.get(key), options[key]);
-    }
-  }
-}
-
-// Mutates axis.labelAnchor, axis.label, scale.label!
-function autoAxisLabelsX(axis, scale, channels) {
-  if (axis.labelAnchor === undefined) {
-    axis.labelAnchor = isOrdinalScale(scale) ? "center" : scaleOrder(scale) < 0 ? "left" : "right";
+export function autoScaleLabels(channels, scales, options) {
+  for (const key in scales) {
+    autoScaleLabel(key, scales[key], channels.get(key), options[key]);
```

Comment:
```suggestion
    autoScaleLabel(key, scales[key], channels.get(key), {label: options.label, ...options[key]});
```

to support the top-level label option (e.g. in the yearly-requests test).

Reply:
It’s slightly more complicated than this because there’s also the facet.label option which overrides the top-level label option for the _fx_ and _fy_ scales (mainly so you can turn them off with label = null).